mempool/src/shared_mempool.rs

```diff
@@ -517,6 +516,57 @@ async fn process_transaction_broadcast<V>(
     }
 }
 
+fn process_broadcast_ack<V>(
+    smp: SharedMempool<V>,
+    peer_id: PeerId,
+    request_id: u64,
+    is_validator: bool,
+) where
+    V: TransactionValidation,
+{
+    let first_txn = smp
+        .peer_info
+        .lock()
+        .expect("[shared mempool] failed to acquire peer_info lock")
+        .get_mut(&peer_id)
+        .expect("[shared mempool] missing peer info for peer")
+        .broadcast_ranges
+        .remove(&request_id);
+    match first_txn {
+        None => error!("[shared mempool] ACK ID not recognized"),
+        Some(first_txn) => {
+            if !is_validator {
+                if request_id > first_txn {
+                    // clean out txns
+                    let mut mempool = smp
+                        .mempool
+                        .lock()
+                        .expect("[shared mempool] failed to acquire mempool lock");
+                    let txns = mempool.range(first_txn, request_id);
+
+                    // helper DS so we call remove_transaction once per sender
+                    let mut max_seq_num = HashMap::new();
```

Comment:
nit: it's cool, but maybe a bit of premature optimization? I'd prefer readability. Let's just remove it